lib/vmpooler/api/v1.rb

```diff
@@ -45,38 +45,88 @@ def alias_deref(hash)
       newhash
     end
 
+    def fetch_single_vm(template)
+      backend.spop('vmpooler__ready__' + template)
+    end
+
+    def return_single_vm(template, vm)
+      backend.spush('vmpooler__ready__' + template, vm)
+    end
+
+    def account_for_starting_vm(template, vm)
+      backend.sadd('vmpooler__running__' + template, vm)
+      backend.hset('vmpooler__active__' + template, vm, Time.now)
+      backend.hset('vmpooler__vm__' + vm, 'checkout', Time.now)
+
+      if Vmpooler::API.settings.config[:auth] and has_token?
+        validate_token(backend)
+
+        backend.hset('vmpooler__vm__' + vm, 'token:token', request.env['HTTP_X_AUTH_TOKEN'])
+        backend.hset('vmpooler__vm__' + vm, 'token:user',
+          backend.hget('vmpooler__token__' + request.env['HTTP_X_AUTH_TOKEN'], 'user')
+        )
+
+        if config['vm_lifetime_auth'].to_i > 0
+          backend.hset('vmpooler__vm__' + vm, 'lifetime', config['vm_lifetime_auth'].to_i)
+        end
+      end
+    end
+
     def checkout_vm(template, result)
-      vm = backend.spop('vmpooler__ready__' + template)
+      vm = fetch_single_vm(template)
 
       unless vm.nil?
-        backend.sadd('vmpooler__running__' + template, vm)
-        backend.hset('vmpooler__active__' + template, vm, Time.now)
-        backend.hset('vmpooler__vm__' + vm, 'checkout', Time.now)
+        account_for_starting_vm(template, vm)
+        update_result_hosts(result, template, vm)
+      else
+        status 503
```

Comment:
I added those comments and will clean them up (thanks for the reminder) ... It looks like vmpooler has always returned a 200 status, even when it cannot allocate vm's, using `return['ok']` as the indicator of whether the allocation was ok.  The 503 status is for cases where no templates were specified.

I don't 100% agree with the way it's using status + `result['ok']`, but I'm going to leave it alone, otherwise this would be a breaking change for folks.